### PR TITLE
Use user-specified conf file path, user and group in puppet::server::conf

### PR DESCRIPTION
**Pull request by tampakrap**

### CODE REVIEW

**Comment by danieldreier** [comment]
Looks reasonable to me. +1 pending CI tests finishing.

File: manifests/server/config.pp [hunk 1]
```diff
@@ -1,10 +1,9 @@
 class puppet::server::config {
 
   include puppet
```
Comment (tampakrap) [inline manifests/server/config.pp]: If I remove this line, two tests fail, can someone please try it and enlighten me why?

Comment (igalic) [inline manifests/server/config.pp]: do your tests make sure that `include puppet` is happening beofre puppet::server::config is processed?
btw, we can ensure this happens by marking this class with [assert_private()](https://github.com/puppetlabs/puppetlabs-stdlib#assert_private)

Comment (tampakrap) [inline manifests/server/config.pp]: correct, so this should stay there

**Comment by igalic** [comment]
thanks @tampakrap!

**Comment by igalic** [comment]
did i just [break the build with this merge](https://travis-ci.org/puppetlabs-operations/puppet-puppet/builds/60193268)?

**Pull request merged**